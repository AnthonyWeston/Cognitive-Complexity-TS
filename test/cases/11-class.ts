class C {
    // nesting 0
}

class D {
    // depth 0
    protected f = () => {
        // depth 0
        while (true);
    }
}

class E {
    // depth 0
    f() {
        // depth 0
        while (true);
    }

    public g() {
        // depth 0
        while (true);
    }
}

const F = class {}

const G = class H {}
