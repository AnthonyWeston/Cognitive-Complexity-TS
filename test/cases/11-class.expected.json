{
    "11-class.ts": {
        "score": 3,
        "inner": [
            {
                "name": "C",
                "score": 0,
                "line": 1,
                "column": 1
            },
            {
                "name": "D",
                "score": 1,
                "line": 5,
                "column": 1,
                "inner": {
                    "name": "f",
                    "score": 1,
                    "line": 7,
                    "column": 5
                }
            },
            {
                "name": "E",
                "score": 2,
                "line": 13,
                "column": 1,
                "inner": [
                    {
                        "name": "f",
                        "score": 1,
                        "line": 15,
                        "column": 5
                    },
                    {
                        "name": "g",
                        "score": 1,
                        "line": 20,
                        "column": 5
                    }
                ]
            }
        ]
    }
}